src/condor_gridmanager/remote_gahp

```diff
@@ -80,10 +86,10 @@ while [ $# != 0 ] ; do
         -* ) break;;
         * )
             if [ "$REMOTE_HOSTNAME" = "" ] ; then
-                REMOTE_HOSTNAME=`echo "$1" | sed 's/.*@//'`
-                echo "$1" | grep -q '@'
-                if [ $? -eq 0 ] ; then
-                    REMOTE_USER=`echo "$1" | sed 's/@.*//'`
+                if  [[ $1 =~ ^(([A-Za-z0-9]*)@)?([A-Za-z0-9\.-]+)(:([0-9]*))?$ ]]; then
```

Comment:
```suggestion
                if  [[ $1 =~ ^(([^@]+)@)?([A-Za-z0-9\.-]+)(:([0-9]*))?$ ]]; then
```
You will want at least 1 character in the user name; also, Unix usernames are fairly permissive.

Reply:
My 2c: usernames should not start with a `-` or digit, nor contain any whitespace or colons. My `useradd` apparently also doesn't allow periods or slashes.  (I haven't tried an exhaustive list of special characters.)

I suspect you'd be good with `[A-Za-z_][A-Za-z0-9_-]*` for the username part.  I feel like `[^@]+` might be a bit more open than you really want.